### PR TITLE
#2536 RP connect command updates and secrets example

### PR DESCRIPTION
**Pull request by asimms41**
## Description

Resolves https://github.com/redpanda-data/documentation-private/issues/2536

Includes:
* Changes to resolve outstanding issues, with streams command, echo and linting subcommands and -c flag.
* Refactoring of secrets example

Review deadline: July 26

### Configuration
* [Main configuration page](https://deploy-preview-18--redpanda-connect.netlify.app/redpanda-connect/configuration/about/)
* [Interpolation](https://deploy-preview-18--redpanda-connect.netlify.app/redpanda-connect/configuration/interpolation/
)
* [Secrets with refactored CLI example](https://deploy-preview-18--redpanda-connect.netlify.app/redpanda-connect/configuration/secrets/#using-cli-flags)

### Guides

* [Streams Mode](https://deploy-preview-18--redpanda-connect.netlify.app/redpanda-connect/guides/streams_mode/about/)

## Checks

- [ ] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [x] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

File: modules/configuration/pages/about.adoc [hunk 1]
```diff
@@ -283,7 +283,7 @@ Echoing is where {page-component-title} can print back your configuration _after
 
 [,bash,subs="attributes+"]
 ----
-rpk connect run -c ./your-config.yaml echo
+rpk connect -c ./your-config.yaml echo
 ----
```
Comment (asimms41) [inline modules/configuration/pages/about.adoc:287]: This command doesn't run with the old or new syntax.

**Comment by Jeffail** [comment]
Hey @asimms41, the new sweeping CLI changes will result in the following command changes:

## `streams`

```
rpk connect -c ./foo.yaml streams ./bar.yaml ./baz.yaml
```
Becomes:
```
rpk connect streams -o ./foo.yaml ./bar.yaml ./baz.yaml
```

## `echo`

```
rpk connect -c ./foo.yaml echo
```
Becomes:
```
rpk connect echo ./foo.yaml
```

## `lint`

```
rpk connect -r ./res/a.yaml lint ./foo.yaml
```
Becomes:
```
rpk connect lint -r ./res/a.yaml ./foo.yaml
```

File: modules/guides/pages/streams_mode/about.adoc [hunk 2]
```diff
@@ -11,13 +11,13 @@ Alternatively, {page-component-title} can be run in `streams` mode, where a sing
...
-When running {page-component-title} in streams mode it is still necessary to provide a general service wide configuration with the `-c`/`--config` flag that specifies observability configuration such as the `metrics`, `logger` and `tracing` sections, as well the `http` section for configuring how the HTTP server should behave.
+When running {page-component-title} in streams mode it is still necessary to provide a general service wide configuration with the `-o`/`--observability` flag that specifies observability configuration such as the `metrics`, `logger` and `tracing` sections, as well the `http` section for configuring how the HTTP server should behave.
 
 You can import resources either in the general configuration, or using the `-r`/`--resources` flag, the same as when running {page-component-title} in regular mode.
 
 [,bash,subs="attributes+"]
 ----
-rpk connect run -r "./prod/*.yaml" -c ./config.yaml streams
+rpk connect run -r "./prod/*.yaml" streams -o ./config.yaml
 ----
```
Comment (asimms41) [inline modules/guides/pages/streams_mode/about.adoc:22]: @Jeffail, I've made some assumptions about the syntax which might incorrect.

File: modules/guides/pages/streams_mode/about.adoc [hunk 3]
```diff
@@ -11,13 +11,13 @@ Alternatively, {page-component-title} can be run in `streams` mode, where a sing
 
 These two methods can be used in combination, i.e. it's possible to update and delete streams that were created with static files.
 
-When running {page-component-title} in streams mode it is still necessary to provide a general service wide configuration with the `-c`/`--config` flag that specifies observability configuration such as the `metrics`, `logger` and `tracing` sections, as well the `http` section for configuring how the HTTP server should behave.
+When running {page-component-title} in streams mode it is still necessary to provide a general service wide configuration with the `-o`/`--observability` flag that specifies observability configuration such as the `metrics`, `logger` and `tracing` sections, as well the `http` section for configuring how the HTTP server should behave.
```
Comment (asimms41) [inline modules/guides/pages/streams_mode/about.adoc:14]: Name for -o flag is just a placeholder for now.

File: modules/guides/pages/streams_mode/about.adoc [hunk 4]
```diff
@@ -11,13 +11,13 @@ Alternatively, {page-component-title} can be run in `streams` mode, where a sing
...
 
 You can import resources either in the general configuration, or using the `-r`/`--resources` flag, the same as when running {page-component-title} in regular mode.
 
 [,bash,subs="attributes+"]
 ----
-rpk connect run -r "./prod/*.yaml" -c ./config.yaml streams
+rpk connect run -r "./prod/*.yaml" streams -o ./config.yaml
```
Comment (Jeffail) [inline modules/guides/pages/streams_mode/about.adoc:20]:
```suggestion
rpk connect streams -r "./prod/*.yaml" -o ./config.yaml
```

Comment (asimms41) [reply modules/guides/pages/streams_mode/about.adoc]: Updated as requested.

File: modules/configuration/pages/secrets.adoc [hunk 5]
```diff
@@ -15,33 +15,52 @@ thing:
...
 
-And we wanted to set the value of `super_secret` to a value stored within something like Hashicorp Vault we could run the config using the `--set` flag with backticks to execute a shell command for the value:
+You could set your credentials to values stored within something like Hashicorp Vault by running the configuration with the `--set` flag and backticks to execute shell commands for the values:
 
 [,bash,subs="attributes+"]
 ----
 rpk connect run -c ./config.yaml \
-  --set "thing.super_secret=`vault kv get -mount=secret thing_secret`"
+  --set "output.aws_dynamodb.credentials.id=`vault kv get -mount=secret access_key_id`" \
+  --set "output.aws_dynamodb.credentials.secret=`vault kv get -mount=secret secret_access_key`"\
```
Comment (Jeffail) [inline modules/configuration/pages/secrets.adoc:57]:
```suggestion
  --set "output.aws_dynamodb.credentials.secret=`vault kv get -mount=secret secret_access_key`"
```

**Review by Jeffail — COMMENTED** [review]
Just one fix suggested, the rest LGTM

File: modules/configuration/pages/secrets.adoc (hunk 5 above)
Comment (asimms41) [reply modules/configuration/pages/secrets.adoc]: Updated as requested.

**Review by Jeffail — APPROVED** [review]
LGTM

**Review by JakeSCahill — APPROVED** [review]

**Pull request merged**